infrastructure/src/main/java/org/wso2/carbon/testgrid/infrastructure/aws/AWSManager.java

```diff
@@ -0,0 +1,198 @@
+     *
+     * @param awsKeyVariableName    Environment variable name for AWS ACCESS KEY.
+     * @param awsSecretVariableName Environment variable name for AWS SECRET KEY.
+     * @throws TestGridInfrastructureException Throws exception when environment variables are not set.
+     */
+    public AWSManager(String awsKeyVariableName, String awsSecretVariableName) throws TestGridInfrastructureException {
+        String awsIdentity = EnvVariableUtil.readEnvironmentVariable(awsKeyVariableName);
+        String awsSecret = EnvVariableUtil.readEnvironmentVariable(awsSecretVariableName);
+        if (StringUtil.isStringNullOrEmpty(awsIdentity) || StringUtil.isStringNullOrEmpty(awsSecret)) {
+            throw new TestGridInfrastructureException("AWS Credentials must be set as environment variables");
+        }
+    }
+
+    /**
+     * This method initiates creating infrastructure through CLoudFormation.
+     *
+     * @param script       Script object containing the CF details.
+     * @param infraRepoDir Path of TestGrid home location in file system as a String.
+     * @return a Deployment object with created infrastructure details.
+     * when an error occurs while reading the cf template file.
+     * @throws TestGridInfrastructureException AWS related error caused by malformed CF template.
+     */
+    public Deployment createInfrastructure(Script script, String infraRepoDir)
+            throws TestGridInfrastructureException {
+        String cloudFormationName = script.getName();
+        AmazonCloudFormation stackbuilder = AmazonCloudFormationClientBuilder.standard()
+                .withCredentials(new EnvironmentVariableCredentialsProvider())
+                .withRegion(infra.getRegion())
+                .build();
+
+        CreateStackRequest stackRequest = new CreateStackRequest();
+        stackRequest.setStackName(cloudFormationName);
+        try {
+            String file = new String(Files.readAllBytes(Paths.get(infraRepoDir, this.infra.getName(),
+                    "AWS", "Scripts", script.getFilePath())));
+            stackRequest.setTemplateBody(file);
+            log.info("Created a CloudFormation Stack with the name :" + stackRequest.getStackName());
+            stackbuilder.createStack(stackRequest);
+            waitForAWSProcess(stackbuilder, cloudFormationName);
+            DescribeStacksRequest describeStacksRequest = new DescribeStacksRequest();
+            describeStacksRequest.setStackName(cloudFormationName);
+            DescribeStacksResult describeStacksResult = stackbuilder.describeStacks(describeStacksRequest);
+            //Initially only PublicDNS which is exported from a dummy script is saved in the Deployment object.
+            List<Host> hosts = new ArrayList<>();
+            for (Stack st : describeStacksResult.getStacks()) {
+                for (Output output : st.getOutputs()) {
+                    if (output.getOutputKey().equals("PublicDNS")) {
```

Comment:
Always compare the variable against the constant value to avoid possible null pointers.
if ("PublicDNS".equals(output.getOutputKey()))